src/components/custom-field-input-number/custom-field-input-number.test.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import renderer from 'react-test-renderer';
+import CustomFieldInputText from '../custom-field-input-text/custom-field-input-text.jsx';
+import CustomFieldInputNumber from './custom-field-input-number.jsx';
+
+describe('CustomFieldInputNumber', () => {
+  it('has defaults', () => {
+    const tree = renderer.create((
+      <CustomFieldInputNumber />
+    )).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  describe('prop-forward API', () => {
+    it('forwards all props accepted by CustomFieldInputText on Object keys', () => {
+      const numberProps = Object.keys(CustomFieldInputNumber.propTypes);
+      Object.keys(CustomFieldInputText.propTypes).forEach((key) => {
+        expect(numberProps).toContain(key);
+      });
+    });
+
+    it('renders all forwarded props except event handlers', () => {
+      const tree = renderer.create((
+        <CustomFieldInputNumber
+          className={'test'}
+          disabled={false}
+          error={true}
+          helpText={'help'}
+          id={'id'}
```

Comment:
Let's avoid having non-unique strings in our values -- especially when we are stringifying the whole JSON tree.

The current `stringTree` has `... id='id' ...` in it and the test asserts the the characters `id` is somewhere in there. If the implementation were to use some other property value than the provided ID then this would be a false positive.

We will want to set the value to something unique and assert that we find it. Or we want to traverse the JSON tree with keys and assert values -- in that case, it is okay to be generic since we will not incorrectly match against a key instead of a value.

Reply:
Yeah, I was taking a shortcut, but I'll make the properties unique and search for the whole property. 😄